lib/rules/color-no-invalid-hex/index.js

```diff
@@ -32,12 +34,16 @@ const rule = (primary) => {
 				return;
 			}
 
+			if (!CONTAINS_HEX.test(decl.value)) {
+				return;
+			}
+
 			valueParser(decl.value).walk(({ value, type, sourceIndex }) => {
 				if (type === 'function' && value.endsWith('url')) return false;
 
 				if (type !== 'word') return;
 
-				const hexMatch = /^#[0-9A-Za-z]+/.exec(value);
+				const hexMatch = /^#[\da-z]+/i.exec(value);
```

Comment:
[ask] Does it make sense to extract a constant like this? Does `new RegExp()` have a performance penalty?

```js
const HEX = /^#[\da-z]+/i;
```

```suggestion
				const hexMatch = new RegExp(HEX).exec(value);
```

Reply:
I am not sure.

I left that one as is because it also has the start anchor `^` and because it uses `exec` which is stateful if I recall correctly.